core/src/main/scala/dagr/core/execsystem/TaskStatusReporter.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.core.execsystem
+
+import java.sql.Timestamp
+
+import dagr.core.tasksystem.Task
+import dagr.core.util.BiMap
+import dagr.core.util.StringUtil._
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/** Provides implements a method to provide an execution report for a task tracker */
+trait TaskStatusReporter extends TaskTracker {
+
+  /** Gets the time stamp as a string (without Nanoseconds), or NA if it is None */
+  private def timestampStringOrNA(timestamp: Option[Timestamp]): String = {
```

Comment:
Putting this here because I wasn't sure where else to put it.  Java 8 introduced a whole new API for dates/times/durations that is, I believe, modeled on Joda Time.  The overview doc is [here](https://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html#package.description).  I'm not sure if you want to take in on as part of this PR, but I think it would be good to switch over to that vs. Timestamp sooner rather than later.